app/services/indexer.rb

```diff
@@ -1,72 +1,68 @@
 # frozen_string_literal: true
 
 class Indexer
-  include Dry::Monads[:result]
+  # @param [RSolr::Client] solr
+  # @param [String] identifier for cocina object
+  # @param [Integer] commit within milliseconds; if nil, then immediately committed.
+  # @raise [Dor::Services::Client::NotFoundResponse]
+  # @raise [Dor::Services::Client::UnexpectedResponse]
+  # @return [Hash,Nil] solr document or nil if indexing failed
+  def self.load_and_index(solr:, identifier:, commit_within: 1000)
```

Comment:
Notice that commit_within is an argument with a default of 1000. This obviates the need for consumers of this class to be passing around additional attributes that are really about setting the commit.